t/t2300-sched-simple.t

```diff
@@ -227,6 +227,26 @@ test_expect_success 'sched-simple: ensure more urgent job run' '
 	EOF
 	test_cmp unlimited-allocs2.expected unlimited-allocs2.out
 '
+# cancel all jobs, to ensure no interference with follow up tests
+# cancel non-running jobs first to ensure they are not accidentally run
+test_expect_success 'sched-simple: cancel jobs' '
+	flux job cancelall --states=SCHED -f &&
+	flux job cancelall -f &&
+	flux job wait-event --timeout=5.0 $(cat job18.id) free &&
+	flux job wait-event --timeout=5.0 $(cat job15.id) free &&
+	flux job wait-event --timeout=5.0 $(cat job16.id) free
+'
+# N.B. Always do the test-free-nolookup tests last, as resources will
+# not actually be freed in the scheduler
+test_expect_success 'sched-simple: reload sched-simple to cover free flags' '
+	flux module reload sched-simple test-free-nolookup
+'
```

Comment:
I think if the alloc response causes a "teardown" as suggested above,  then the resources will not be leaked as long as the scheduler is reloaded again afterwards without the test option.   The free should be resent to the new scheduler instance.